src/views/Home.vue

```diff
@@ -0,0 +1,369 @@
+    },
+
+    getProduct(productId: string) {
+      return this.products[productId] ? this.products[productId] : {}
+    },
+
+    async saveShippingAddressChanges(shipGroup: any) {
+      let resp
+      const payload = {
+        "orderId": this.order.id,
+        "shipGroupSeqId": shipGroup.shipGroupSeqId,
+        "contactMechId": shipGroup.shipTo.postalAddress.id,
+        "shipmentMethod": `${this.deliveryMethod}@_NA_`,
+        "contactMechPurposeTypeId": "SHIPPING_LOCATION",
+        "facilityId": "WH",
+        "toName": `${shipGroup.editedShipmentAddress.firstName} ${shipGroup.editedShipmentAddress.lastName}`,
+        "address1": shipGroup.editedShipmentAddress.address1,
+        "city": shipGroup.editedShipmentAddress.city,
+        "stateCode": shipGroup.editedShipmentAddress.stateCode,
+        "postalCode": shipGroup.editedShipmentAddress.postalCode,
+        "country": shipGroup.editedShipmentAddress.country
+      } as any
+
+      if (shipGroup.selectedShipmentMethodTypeId === shipGroup.shipmentMethodTypeId) {
+        // In case of address edit, we honour the previously selected delivery method
+        payload.shipmentMethod = `${shipGroup.shipmentMethodTypeId}@_NA_`
+        payload.isEdited = true
+      }
+
+      try {
+        resp = await OrderService.updateShippingAddress(payload);
+        if (resp.status === 200 && !hasError(resp) && resp.data) {
+          this.areChangesSaved = true;
+          showToast(translate("Changes saved"))
+        } else {
+          showToast(translate("Failed to update the shipping addess"))
+        }
+      } catch (error) {
+        console.error(error)
+        showToast(translate("Failed to update the shipping addess"))
+      }
+    },
+
+    async saveFacilityChanges(shipGroup: any) {
+      let resp
+      const payload = {
+        "orderId": this.order.id,
+        "shipGroupSeqId": shipGroup.shipGroupSeqId,
+        "contactMechId": shipGroup.shipTo.postalAddress.id,
+        "shipmentMethod": `STOREPICKUP@_NA_@CARRIER`,
+        "contactMechPurposeTypeId": "SHIPPING_LOCATION",
+        "facilityId": shipGroup.selectedFacilityId,
+      }
+
+      try {
+        resp = await OrderService.updateFacility(payload);
+        if (resp.status === 200 && !hasError(resp)) {
+          this.areChangesSaved = true;
+          showToast(translate("Changes saved"))
+        } else {
+          showToast(translate("Failed to update the pickup store"))
+        }
+      } catch (error) {
+        console.error(error)
+        showToast(translate("Failed to update the pickup store"))
+      }
+    },
+
+    async saveShipmentChanges(shipGroup: any) {
+      shipGroup.selectedShipmentMethodTypeId === 'STOREPICKUP' ? this.saveFacilityChanges(shipGroup) : this.saveShippingAddressChanges(shipGroup);   
+    },
+
+    updateDeliveryMethod(event: any, shipGroup: any) {
+      this.order.shipGroup.map((group: any) => {
+        if (group.shipGroupSeqId === (shipGroup as any).shipGroupSeqId) {
```

Comment:
I think this explicit casting is not needed here.
```suggestion
        if (group.shipGroupSeqId === shipGroup.shipGroupSeqId) {
```